packages/eas-cli/src/uploads.ts

```diff
@@ -31,6 +32,37 @@ export async function uploadWithPresignedPostAsync(
   file: string,
   presignedPost: PresignedPost,
   handleProgressEvent?: ProgressHandler
+): Promise<Response> {
+  return await promiseRetry(
+    async retry => {
+      let response: Response;
+      try {
+        response = await uploadWithPresignedPostInternalAsync(
+          file,
+          presignedPost,
+          handleProgressEvent
+        );
+      } catch (e: any) {
+        return retry(e);
+      }
+
+      if (!response.ok) {
+        return retry(new Error(`Presigned post responded with a ${response.status}`));
+      }
+
+      return response;
+    },
+    {
+      retries: 3,
+      factor: 2,
+    }
```

Comment:
May want to link to https://cloud.google.com/storage/docs/retry-strategy#node.js saying the parameters were chosen by matching GCS's defaults